native/core/src/activities.rs

```diff
@@ -123,9 +132,34 @@ impl Activity {
 
     pub fn type_str(&self) -> String {
         match &self.inner {
-            ActivityContent::MembershipChange(c) => c.as_str(),
-            ActivityContent::RoomCreate(_) => "roomCreate",
-            ActivityContent::RoomName(_) => "roomName",
+            ActivityContent::ProfileChange { .. } => "profileChange",
+            ActivityContent::MembershipChange { .. } => "membershipChange",
```

Comment:
I don't like this change. The more specific `left` type has been wrapped into a more opaque "membershipChange" type that the UI then first has to figure out again that this was actually someone leaving rather than just updating their display name. This is inconvenient for the UI to build around. In general we took the approach to be as specific as possible as highly in the layers as possible, so this should return `left` and then can hold internally the `membershipChange`-object that is handed out for details, but it shouldn't be required to go that route.